lib/measures/tbd/resources/psi.rb

```diff
@@ -177,6 +177,50 @@ def complete?(s) # true/false
   end
 end
 
+##
+# Check for matching vertex pairs between edges (1" tolerance).
+# @param [Hash] e1 First edge
+# @param [Hash] e2 Second edge
+#
+# @return [Boolean] Returns true if edges share vertex pairs
+def matches?(e1, e2)
+  raise "matches? missing edges" unless e1 && e2
+  raise "matches? missing :v0 for e1" unless e1.has_key?(:v0)
+  raise "matches? missing :v1 for e1" unless e1.has_key?(:v1)
+  raise "matches? missing :v0 for e2" unless e2.has_key?(:v0)
+  raise "matches? missing :v1 for e2" unless e2.has_key?(:v1)
+  raise "e1 v0: #{e1[:v0].class}? expected a Topolys point3D" unless e1[:v0].is_a?(Topolys::Point3D)
+  raise "e1 v1: #{e1[:v1].class}? expected a Topolys point3D" unless e1[:v1].is_a?(Topolys::Point3D)
+  raise "e2 v0: #{e1[:v0].class}? expected a Topolys point3D" unless e2[:v0].is_a?(Topolys::Point3D)
+  raise "e2 v1: #{e1[:v1].class}? expected a Topolys point3D" unless e2[:v1].is_a?(Topolys::Point3D)
+
+  answer = false
+  answer = true if
+  (
```

Comment:
I think this would return true for the degenerate cases:

e1 = [(1,0,0), (1,0,0)] and e2 = [(1,0,0), (2,0,0)]

e1 = [(1,0,0), (2,0,0)] and e2 = [(1,0,0), (1,0,0)]

Reply:
Thanks. I'll raise (invalid) cases where an edge has a length of ´0.